src/geom/prep/AbstractPreparedPolygonContains.cpp

```diff
@@ -112,9 +112,11 @@ AbstractPreparedPolygonContains::eval(const geom::Geometry* geom)
     //
     // If a point of any test components does not lie in target,
     // result is false
-    bool isAllInTargetArea = isAllTestComponentsInTarget(geom);
-    if(!isAllInTargetArea) {
-        return false;
+    if (!requireSomePointInInterior || geom->getGeometryTypeId() != GEOS_POINT) {
```

Comment:
This new test is kinda hard to comprehend, when combined with following negative test.  Am trying to think of how this can be rephrased to be .a bit more intuitive.  

At least a comment would help

Reply:
@dr-jts Agreed; I reworked it in a2030535758d04aa64e6